app/src/main/java/com/example/gfood/data/source/remote/CategoryRemoteDataSource.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.gfood.data.source.remote
+
+import com.example.gfood.data.model.Category
+import com.example.gfood.data.source.CategoryDataSource
+import com.example.gfood.data.source.remote.fetchjson.GetJsonFromUrl
+import com.example.gfood.utils.Constant
+import com.example.gfood.utils.KeyEntity
+
+@Suppress("DEPRECATION")
+class CategoryRemoteDataSource : CategoryDataSource.Remote {
+    private var baseUrl = Constant.BASE_URL + Constant.BASE_CATEGORY
+
+    private object Holder {
+        val INSTANCE = CategoryRemoteDataSource()
+    }
+
+    override fun getCategory(listener: OnFetchDataJsonListener<MutableList<Category>>) {
+        GetJsonFromUrl(KeyEntity.CATEGORY, listener).execute(baseUrl)
+    }
+
+    companion object {
+        val instance: CategoryRemoteDataSource by lazy { Holder.INSTANCE }
+    }
+}
```

Comment:
Tất cả những class thiết kế singleton em thiết kế theo mẫu sau nhé 
Singleton Private constructor 
```suggestion
class Singleton private constructor(
    private val param1: Param1, 
    private val param2: Param2
) {

    companion object {
        private var instance : Singleton? = null
        
        fun getInstance(param1: Param1, param2: Param2): Singleton {
            return instance ?: Singleton(param1, param2).also{
                instance = it
            }
        }
    }
}
```

Reply:
Dạ em fix rồi á anh